crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -28,21 +28,29 @@ pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
     #[serde(with = "alloy_serde::u64_hex_opt")]
     pub block_number: Option<u64>,
     /// Gas used by this transaction alone.
-    #[serde(with = "alloy_serde::u64_hex_opt")]
-    pub gas_used: Option<u64>,
+    #[serde(with = "alloy_serde::u128_hex_or_decimal_opt")]
+    pub gas_used: Option<u128>,
     /// The price paid post-execution by the transaction (i.e. base fee + priority fee). Both
     /// fields in 1559-style transactions are maximums (max fee + max priority fee), the amount
     /// that's actually paid by users can only be determined post-execution
-    #[serde(with = "alloy_serde::u64_hex")]
-    pub effective_gas_price: u64,
+    #[serde(with = "alloy_serde::u128_hex_or_decimal")]
```

Comment:
only hex for all rpc fields

Reply:
Yes, I believe there should be a separate PR for #465 as it's likely to break several tests, and I also wanted to unblock https://github.com/foundry-rs/foundry/pull/7106 quickly. I'll implement it immediately after this.